src/error.rs

```diff
@@ -0,0 +1,33 @@
+use defines::AfError;
+
+pub type ErrorCallback = Fn(AfError);
+
+pub static mut HANDLE_ERROR: &'static ErrorCallback = &handle_error_general; 
+
+pub fn register_error_handler(callback: &'static ErrorCallback) {
+    unsafe {
+        HANDLE_ERROR = callback;
+    }
+}
+
+pub fn handle_error_general(error_code: AfError) {
+    match error_code {
+        AfError::SUCCESS            => {}, /* No-op */
```

Comment:
Suggestion: use the function (maybe move it outside of the trait impl) to get the `&str` 's here to avoid having code dupes:
```rust
impl Error for AfError {
    fn description(&self) -> &str {
```